cpp/src/arrow/compute/kernels/hash.cc

```diff
@@ -331,8 +328,7 @@ class RegularHashKernelImpl : public HashKernelImpl {
   }
 
   template <typename Enable = Status>
-  auto VisitValue(const Scalar& value) ->
-      typename std::enable_if<!with_error_status, Enable>::type {
+  auto VisitValue(const Scalar& value) -> enable_if_t<!with_error_status, Enable> {
```

Comment:
The signature is poorly formed: The enabling condition does not depend on a deduced template parameter so SFINAE can't happen. It's probably not disabling what it was intended to disable

Reply:
Can you check the new version?